packages/core/src/components/fields/MultiSchemaField.tsx

```diff
@@ -237,6 +237,7 @@ class AnyOfField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends For
             autofocus={autofocus}
             label={title ?? name}
             hideLabel={!displayLabel}
+            readonly={this.props.readonly}
```

Comment:
Add the destructure of `readonly` from `props` above on line 155 and just use:
```suggestion
            readonly={readonly}
```